woocommerce/woocommerce-ios
# [Scan to Update Inventory M1] Track events

Closes: #11440 

## Description:
This PR adds track events for Scan to Update Inventory M1. Described in pdfdoF-4k6-p2

## Changes:
* Added new `WooAnalyticsStat` cases
* Updated `WooAnalyticsEvent.BarcodeScanning.Source` with new `scan_to_update_inventory` property
* Updated existing `productSearchViaSKUSuccess` event with an additional optional property: `stock_managed: true/false`
* Injected `Analytics` service in `UpdateProductInventoryViewModel`, from which we call the different events to be tracked.
* Added Unit Tests

## Testing instructions
* Tap the scan button, see `product_list_product_barcode_scanning_tapped`
* Successfully scan a product, see `product_search_via_sku_success` with properties `source: scan_to_update_inventory`, and `stock_managed: true/false`
* Tap on `View Product Details`, see `product_quick_inventory_view_product_details_tapped`
* Tap `Quantity +1`, see:
  * `product_quick_inventory_update_increment_quantity_tapped`
  * `product_quick_inventory_quantity_update_success`, or `product_quick_inventory_quantity_update_failure` if we force an error.
* Successfully scan a product again, update the quantity manually, then tap the `Update Quantity` button, see:
  * `product_quick_inventory_update_manual_quantity_update_tapped``
  * `product_quick_inventory_quantity_update_success`, or `product_quick_inventory_quantity_update_failure` if we force an error.
* Successfully scan a product again, tap on "Done" without saving any changes, see `product_quick_inventory_update_dismissed`

**Stock Management:**
* Disable stock management on a product, see `product_search_via_sku_success` with properties `source: scan_to_update_inventory`, and `stock_managed: true`
* Tap on `Manage Stock`, see `product_quick_inventory_enable_manage_stock_success`
* Force an error when tapping `Manage Stock`, see `product_quick_inventory_enable_manage_stock_failure` with error description.
